src/movies.js

```diff
@@ -1,16 +1,117 @@
-// Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+/*  Iteration 1: All directors? - Get the array of all directors.  */
+/*  Bonus: It seems some of the directors had directed multiple movies,
+    so they will pop up multiple minutess in the array of directors.
+    How could you "clean" a bit this array and make it unified (without duplicates)? */
+const getAllDirectors = (movies) => {
+  return movies.map((movie) => movie.director);
+};
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+/*  Iteration 2: How many drama movies did Steven Spielberg direct?  */
+const howManyMovies = (movies) => {
+  return movies.filter((movie) => {
+    return (
+      movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+    );
+  }).length;
+};
 
-// Iteration 3: All rates average - Get the average of all rates with 2 decimals
+/*  Iteration 3: All rates average - Get the average of all rates with 2 decimals.  */
+const ratesAverage = (movies) => {
+  const avgRating = movies.reduce((acc, movie, _, src) => {
+    return acc + (movie.rate || 0) / src.length;
```

Comment:
amazing use of the or operator here. works like a charm.

Reply:
Also, perfect dividing every item by the length inside the reduce instead of one time the whole sum between the number of movies. some little cool math trick in there.